src/util/promiseSequence.ts

```diff
@@ -0,0 +1,10 @@
+export default function sequence<T>(funcs: (() => Promise<T>)[]): Promise<T[]> {
+  return funcs.reduce((prom: Promise<T[]>, func: () => Promise<T>): Promise<
+    T[]
+  > => {
+    return prom.then(
```

Comment:
Lack of (?) catch / error handling makes me really nervous

Reply:
As per my other comment, this is consumed by scheduler in `src/worker.ts`; it is the responsibility of the scheduler to handle the execution errors at a global level rather than within the task. I.e. we intentionally want the error to bubble up to the caller.

Reply:
Ok I see. I'm not seeing an example of error handling in the tests - would be nice to have since this whole file is dedicated to proper promise handling.

Reply:
Thanks for this, you made me think a bit more about the intented behaviour of this lib. I've now added a `catch` to swallow any promise errors and continue processing with the chain.